src/objects.c

```diff
@@ -4107,3 +4115,11 @@ P11PROV_OBJ *mock_pub_ec_key(P11PROV_CTX *ctx, CK_ATTRIBUTE_TYPE type,
 
     return key;
 }
+
+P11PROV_OBJ *p11prov_obj_pub_from_generated(P11PROV_OBJ *obj)
+{
+    if (obj == NULL || obj->isGenerated == false) {
```

Comment:
Do we need to limit this functionality only to generated keys? I think it might be also useful for non-generated ones as the EVP_PKEY private key is mostly abstraction of a key pair in OpenSSL so searching for associated objects to do a verification/decryption would make sense to me also for existing key objects.

Reply:
I think we should refactor the code to have a obj->public pointer to another P11PROV_OBJ, so that we can keep the object on generation or dynamically load the associated public key (if possible) as need.
We currently load the associated public key explicitly in just some places, but we should generalize this so more parts will work correctly automatically.

We may also have to set a sentinel value to *prevent* trying to load/expecting a public key if OpenSSL explicitly imported into the provider a key without public bits, and/or to signal that we already searched once and no associated public key was found ion the token.

Reply:
> I think we should refactor the code to have a obj->public pointer to another P11PROV_OBJ

Done

> We may also have to set a sentinel value to prevent trying to load/expecting a public key

Public key object is set only when generating key pairs, otherwise it will be null

Reply:
I will handle his later on my own, my comment were assuming a change where we search in the token if we are asked for a public key and the pointer is NULL.